event-buffer/event-buffer-core/src/main/java/uk/gov/justice/services/event/buffer/core/repository/service/ConsecutiveEventBufferService.java

```diff
@@ -0,0 +1,135 @@
+import java.util.stream.StreamSupport;
+
+import javax.annotation.Priority;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Alternative;
+import javax.inject.Inject;
+
+import static java.lang.String.format;
+import static uk.gov.justice.services.messaging.logging.JsonEnvelopeLoggerHelper.toEnvelopeTraceString;
+
+@ApplicationScoped
+@Alternative
+@Priority(2)
+public class ConsecutiveEventBufferService implements EventBufferService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventBufferService.class);
+
+    @Inject
+    private StreamBufferJdbcRepository streamBufferRepository;
+
+    @Inject
+    private StreamStatusJdbcRepository streamStatusRepository;
+
+    @Inject
+    private JsonObjectEnvelopeConverter jsonObjectEnvelopeConverter;
+
+
+    /**
+     *
+     * Takes an incoming event and returns a stream of json envelopes. If the event is not consecutive
+     * according to the stream_status repository then an empty stream is returned and the incomingEvent
+     * is added to the event-buffer-repository. If the incomingEvent is consecutive then it is returned
+     * as a stream with any consecutive events from the buffer. If an event is the first to be processed
+     * for that streamId then the version value must be 1 or the incomingEvent is added to the buffer
+     * and an empty stream is returned.
+     *
+     * @param incomingEvent
+     * @return streamOfConsecutiveEvents
+     */
+    @Override
+    public Stream<JsonEnvelope> currentOrderedEventsWith(final JsonEnvelope incomingEvent) {
+
+        LOGGER.trace(format("Message buffering for message: %s", toEnvelopeTraceString(incomingEvent)));
+
+        final UUID streamId = incomingEvent.metadata().streamId().get();
+        final Long incomingEventVersion = incomingEvent.metadata().version().get();
+
+        final Optional<StreamStatus> currentStatus = streamStatusRepository.findByStreamId(streamId);
+        if (currentStatus.isPresent()) {
+            final Long currentVersion = currentStatus.get().getVersion();
+            if (incomingEventObsolete(incomingEventVersion, currentVersion)) {
+                LOGGER.trace(format("Message : %s is an obsolete version", toEnvelopeTraceString(incomingEvent)));
+                return Stream.empty();
+
+            } else if (incomingEventNotInOrder(incomingEventVersion, currentVersion)) {
+                LOGGER.trace(format("Message : %s is not consecutive, adding to buffer", toEnvelopeTraceString(incomingEvent)));
+                return addToBufferAndReturnEmptyStream(incomingEvent, streamId, incomingEventVersion);
+
+            } else {
+                LOGGER.trace(format("Message : %s version is valid sending stream to dispatcher", toEnvelopeTraceString(incomingEvent)));
+                return streamOfIncomingAndBufferedEvents(incomingEvent, streamId, incomingEventVersion);
+
+            }
+        } else {
+            if(incomingEventVersion == 1) {
```

Comment:
Can we make this a constant 'FIRST_VERSION' or 'INITIAL_VERION' etc. And we're sure we start on 1 not 0? :)